index.js

```diff
@@ -54,36 +70,48 @@ module.exports = {
       }
       if (!initSpec(content)) return cb(new Error(initSpec.errorsString))
 
-      const recipientKeys = [
-        { key: groupKey.toBuffer(), scheme: keySchemes.private_group },
-      ]
-
-      ssb.db.create(
-        {
-          content,
-          recps: recipientKeys,
-          encryptionFormat: 'box2',
-        },
-        (err, groupInitMsg) => {
-          if (err) return cb(err)
+      const groupKey = new SecretKey()
 
-          const data = {
-            id: buildGroupId({ groupInitMsg, groupKey: groupKey.toBuffer() }),
-            secret: groupKey.toBuffer(),
-            root: groupInitMsg.key,
-            subfeed: '',
-          }
+      findOrCreateGroupFeed(groupKey, (err, groupFeed) => {
+        if (err) return cb(err)
 
-          ssb.box2.addGroupInfo(data.id, { key: data.secret, root: data.root })
+        const recps = [
+          { key: groupKey.toBuffer(), scheme: keySchemes.private_group },
+        ]
 
-          // adding myself for recovery reasons
-          addMembers(data.id, [ssb.id], {}, (err) => {
+        ssb.db.create(
+          {
+            keys: groupFeed.keys,
+            content,
+            recps,
+            encryptionFormat: 'box2',
+          },
+          (err, groupInitMsg) => {
             if (err) return cb(err)
 
-            return cb(null, data)
-          })
-        }
-      )
+            const data = {
+              id: buildGroupId({ groupInitMsg, groupKey: groupKey.toBuffer() }),
+              secret: groupKey.toBuffer(),
+              root: groupInitMsg.key,
+              subfeed: groupFeed.keys,
+            }
+
+            ssb.box2.addGroupInfo(data.id, {
+              key: data.secret,
```

Comment:
`<aside>` if we're breaking the API of box2, I would love to rename the groupKey to `secret` or `groupSecret` everywhere... or as many places as possible. I really regret the over-use of the word "Key"